pkg/distributor/receiver/middleware.go

```diff
@@ -0,0 +1,74 @@
+package receiver
+
+import (
+	"context"
+
+	"github.com/weaveworks/common/user"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/grafana/loki/pkg/util/log"
+)
+
+type ConsumeLogsFunc func(context.Context, pdata.Logs) error
+
+func (f ConsumeLogsFunc) Capabilities() consumer.Capabilities {
```

Comment:
This is pretty much the default implementation. If you use the `consumerhelper.NewLogs`, you can just pass the ConsumeLogsFunc to it (even as a private func) instead:

https://github.com/open-telemetry/opentelemetry-collector/blob/839909863acceff104c18a9db1b5d24c13d6a972/consumer/consumerhelper/logs.go#L37-L46

Reply:
consumerhelper.NewLogs.does not exist

![image](https://user-images.githubusercontent.com/9583245/154189615-54e959de-d017-49d5-9b3a-f1be08032e21.png)

Reply:
I don't know what to say :-) The code seems to be there. Don't you need to update your vendor?

https://github.com/open-telemetry/opentelemetry-collector/blob/v0.44.0/consumer/consumerhelper/logs.go

Reply:
thanks，it looks like there is something wrong with my go vendor, I will try to fix it